From: Mike Snitzer <snitzer@redhat.com>
Date: Thu, 31 Mar 2016 16:04:25 -0400
Subject: dm mpath: move trigger_event member to the end of 'struct multipath'
References: bnc#974525,FATE#320741
Git-commit: 20800cb3450ee44ec1827d7e8bbfd5a9dc02e6cd
Patch-Mainline: v4.7-rc1

Allows the 'work_mutex' member to no longer cross a cacheline.

Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/dm-mpath.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/md/dm-mpath.c b/drivers/md/dm-mpath.c
index 780e5d0..54daf96 100644
--- a/drivers/md/dm-mpath.c
+++ b/drivers/md/dm-mpath.c
@@ -89,8 +89,6 @@ struct multipath {
 	atomic_t pg_init_in_progress;	/* Only one pg_init allowed at once */
 	atomic_t pg_init_count;		/* Number of times pg_init called */
 
-	struct work_struct trigger_event;
-
 	/*
 	 * We must use a mempool of dm_mpath_io structs so that we
 	 * can resubmit bios on error.
@@ -98,6 +96,7 @@ struct multipath {
 	mempool_t *mpio_pool;
 
 	struct mutex work_mutex;
+	struct work_struct trigger_event;
 };
 
 /*
-- 
1.8.5.6

